src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -147,40 +175,21 @@ internal JsonTypeInfo(Type type, JsonSerializerOptions options!!, bool dummy)
             PropertyInfoForTypeInfo = null!;
         }
 
-        [RequiresUnreferencedCode(JsonSerializer.SerializationUnreferencedCodeMessage)]
-        internal JsonTypeInfo(Type type, JsonSerializerOptions options) :
-            this(
-                type,
-                GetConverter(
-                    type,
-                    parentClassType: null, // A TypeInfo never has a "parent" class.
-                    memberInfo: null, // A TypeInfo never has a "parent" property.
-                    options),
-                options)
-        {
-        }
-
-        [RequiresUnreferencedCode(JsonSerializer.SerializationUnreferencedCodeMessage)]
         internal JsonTypeInfo(Type type, JsonConverter converter, JsonSerializerOptions options)
         {
             Type = type;
             Options = options;
-            NumberHandling = GetNumberHandlingForType(Type);
             PropertyInfoForTypeInfo = CreatePropertyInfoForTypeInfo(Type, converter, NumberHandling, Options);
```

Comment:
Did you try removing `PropertyInfoForTypeInfo` in the end?

Reply:
I did partially (I finished only deserialization part) but this will be super risky change and it can possibly regress perf. I'm not planning to retry the experiment until I finish the contract resolving because this requires ground up refactoring.

Reply:
Our tolerance for refactoring infra/risky changes is bound to be reduced as we get closer to RC. It's the kind of stuff where it's better to do early. In other words, it's fine to keep as-is if it's not easy to change.